### PR TITLE
tests: Skip unnecessary sleeps in cql_test_env teardown

### PR DESCRIPTION
**Pull request by tgrabiec**
This PR contains two patches which get rid of unnecessary sleeps on cql_test_env teardown greatly reducing run time of tests.

Reduces run time of `build/dev/test/boost/schema_change_test` from 90s to 6s.

### CODE REVIEW

File: test/lib/cql_test_env.cc [hunk 1]
```diff
@@ -944,6 +944,10 @@ class single_node_cql_env : public cql_test_env {
             with_scheduling_group(dbcfg.statement_scheduling_group, [&func, this] {
                 return func(*this);
             }).get();
+
+            abort_sources.invoke_on_all([] (auto&& as) {
+                as.request_abort();
```
Comment (kbr-scylla) [inline test/lib/cql_test_env.cc:949]: Won't this interrupt clean shutdown of services, causing warnings or errors?

File: test/lib/cql_test_env.cc [hunk 2]
```diff
@@ -944,6 +944,10 @@ class single_node_cql_env : public cql_test_env {
             with_scheduling_group(dbcfg.statement_scheduling_group, [&func, this] {
```
Comment (kbr-scylla) [inline test/lib/cql_test_env.cc:944]: > Before this patch, there was a 1s sleep on gossiper shutdown, which I
don't know where it comes from. After the patch there is no such
sleep.

The frequency of main gossiper loop is once per second
```
    static constexpr std::chrono::milliseconds INTERVAL{1000};
```

maybe we're waiting for the last gossiper loop iteration to execute?

**Review by kbr-scylla — COMMENTED** [review]
> Reduces run time of build/dev/test/boost/schema_change_test from 90s to 6s.

lolwut

**Comment by avikivity** [comment]
> > Reduces run time of build/dev/test/boost/schema_change_test from 90s to 6s.
> 
> lolwut

That's in redis/lolwut.cc

File: test/lib/cql_test_env.cc (hunk 2 above)
Comment (avikivity) [reply test/lib/cql_test_env.cc:944]: likely

**Review by avikivity — APPROVED** [review]

**Review by xemul — APPROVED** [review]

File: test/lib/cql_test_env.cc [hunk 3]
```diff
@@ -675,6 +675,7 @@ class single_node_cql_env : public cql_test_env {
             gcfg.cluster_name = "Test Cluster";
             gcfg.seeds = std::move(seeds);
             gcfg.skip_wait_for_gossip_to_settle = 0;
+            gcfg.shutdown_announce_ms = 0;
```
Comment (nyh) [inline test/lib/cql_test_env.cc:678]: Amazing!

While I kept optimizing test/cql-pytest/run to get a test running and shutting down in 0.3 seconds, I forgot that this should also be done in the boost tests... Imagine how many seconds we wasted in the last 6 years running those tests :-(

**Review by nyh — APPROVED** [review]

File: test/lib/cql_test_env.cc (hunk 1 above)
Comment (kbr-scylla) [reply test/lib/cql_test_env.cc:949]: > it may do things like abort a compaction, but who cares

heh

**Comment by kbr-scylla** [comment]
I queued a revert of the second commit: c3257bf546a5da5007ad6b167c35fe5b22630c6b
If I understand correctly, majority of gains come from the first commit.

**Pull request merged**